providers/class.two-factor-fido-u2f.php

```diff
@@ -55,7 +147,190 @@ public function validate_authentication( $user ) {}
 	 * @since 0.1-dev
 	 *
 	 * @param WP_User $user WP_User object of the logged-in user.
+	 * @return boolean
+	 */
+	public function is_available_for_user( $user ) {
+		return self::is_browser_support() && (bool) self::get_security_keys( $user->ID );
+	}
+
+	/**
+	 * Inserts markup at the end of the user profile field for this provider.
+	 *
+	 * @since 0.1-dev
+	 *
+	 * @param WP_User $user WP_User object of the logged-in user.
+	 */
+	public function user_options( $user ) {
+		?>
+		<div>
+			<?php echo esc_html( __( 'You need to register security keys such as Yubikey.' ) ); ?>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Add registered security key to a user.
+	 *
+	 * @since 0.1-dev
+	 *
+	 * @param int    $user_id  User ID.
+	 * @param object $register The data of registered security key.
+	 * @return int|bool Meta ID on success, false on failure.
 	 */
-	public function is_available_for_user( $user ) {}
+	public static function add_security_key( $user_id, $register ) {
+		if ( ! is_numeric( $user_id ) ) {
+			return false;
+		}
 
+		if (
+			! is_object( $register )
+				|| ! property_exists( $register, 'keyHandle' ) || empty( $register->keyHandle )
+				|| ! property_exists( $register, 'publicKey' ) || empty( $register->publicKey )
+				|| ! property_exists( $register, 'certificate' ) || empty( $register->certificate )
+				|| ! property_exists( $register, 'counter' ) || ( -1 > $register->counter )
+		) {
+			return false;
+		}
+
+		$register = array(
+			'keyHandle'   => $register->keyHandle,
+			'publicKey'   => $register->publicKey,
+			'certificate' => $register->certificate,
+			'counter'     => $register->counter,
+		);
+
+		$register['name']      = 'New Security Key';
```

Comment:
Should this name be made available for translation?  Is it user editable anywhere?